native/src/main/java/io/ballerina/stdlib/tcp/TcpListener.java

```diff
@@ -90,7 +90,8 @@ protected void initChannel(SocketChannel channel) throws Exception {
                         callback.complete(null);
                     } else {
                         if (!isCallbackCompleted.get()) {
-                            callback.complete(Utils.createTcpError("Error initializing the server."));
+                            callback.complete(Utils.createTcpError("Error initializing the server: "
+                                    + channelFuture.cause().getMessage()));
```

Comment:
Shall we use `String.format` to create the error message ?
Check for other places as well.
```suggestion
                            String errorMsg = String.format("Error initializing the server:%s", channelFuture.cause().getMessage());
                            callback.complete(Utils.createTcpError(errorMsg));
```